qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
+    """Parse the given numbers set in text format, and return it as a list of
+    numbers.
+
+    Args:
+        txtset: Numbers set specification. E.g. '1,5,8-last,32,2', where 'last'
+                is a special keyword that denotes that maximum allowable
+                number.
+        nummax: The number that will be used to replace 'last'
+    """
+    txtset = txtset.lower().replace('last', '%d' % (nummax))
+    l = set()
+    parts = txtset.split(',')
+    for part in parts:
+        if part.find('-') != -1:
+            [start, end] = part.split('-')
+            start = int(start)
+            end = int(end)
+            for i in range(int(start), int(end)+1):
+                l.add(i)
+        else:
+            l.add(int(part))
+    return sorted(l)
+
+
+def parse_list_into_numsettxt(l):
```

Comment:
Currently, my issue is that, with my previous function `parse_numsettext_into_list(...)`, someone could do things like `:download-delete 1-123443`, and if we go with the list one, then the error will be like "Downloads 5,6,7,8,9,10,...,123443 don't exist" (without "...").

That's why I put this function, to undo that crazy long list into "5-123443" (assuming downloads 1,2,3,4 were deletable).

The asymptotic worst run time complexity of this function is obviously O(len(numbers)). Maybe by itself is not a problem, but it can be a very large number depending on whether someone defines a crazy number set. For example, if someone does `:download-delete 1-1234325`, then the len(numbers) = 1234321 (if 5 got deleted).